### DISCUSSION TITLE
vertico buffer mode slow when font size is decreased

### DISCUSSION THREAD

**Discussion by rougier** [minad/vertico#427] [Q&A] [answered]
I've observed that when I decreased font size while in minibuffer (using `C-x C--`) with vertico-buffer-mode active, it makes navigation/typing to become really slow (it is not the case for increased font size). I would thus like to know if it is the case for everyone or only me.

**Comment by minad** [comment]
I don't see any slow down after I did `M-x C-x C-- - - -` on Emacs 29.1 Linux/X11. You can capture a profile and check where the slow down is coming from.

EDIT: I missed that you specifically mentioned `vertico-buffer-mode`. I observe a horrible slow down there too.

**Reply by minad** [reply]
Looks like a problem in the display engine in relation with face mapping, at least `redisplay` appears prominently in the profile. I suggest to avoid `text-scale-adjust`. I've seen moderate slow down caused by it before in other buffers, but not as bad as this one.

**Reply by minad** [reply, marked as answer]
Investigating a little bit more, this seems to be caused by the pre-redisplay function of vertico-buffer. At least the problem goes away when I remove this hook. I have to check if I am doing something problematic there which somehow causes a recursion of the redisplay code. Ideally such pre-redisplay functions are completely avoided, but I guess I didn't find a better solution back then when I wrote this.

EDIT: Worked around the problem in https://github.com/minad/vertico/commit/37e01c10bf38fd154d90ec315aaf6414a046879d. This is definitely a problem in the display engine, where we hit a slow path for large vscroll values.

**Reply by rougier** [reply]
Thanks, I confirm it fixed the problem.
